src/Stratis.Features.FederatedPeg/Collateral/CheckCollateralFullValidationRule.cs

```diff
@@ -45,10 +53,36 @@ public override Task RunAsync(RuleContext context)
 
             IFederationMember federationMember = this.slotsManager.GetFederationMemberForTimestamp(context.ValidationContext.BlockToValidate.Header.Time);
 
-            if (!this.collateralChecker.CheckCollateral(federationMember))
+            byte[] rawCommitmentData = this.encoder.ExtractRawCommitmentData(context.ValidationContext.BlockToValidate.Transactions.First());
```

Comment:
I think it's probably best to explicitly check that the first transaction is a coinbase, so that we don't implicitly rely on other rules doing that.

Reply:
Why? We have a rule that checks exactly that and it is executed before the `CheckCollateralFullValidationRule` so we have a guarantee that it exists at this point